pkg/ruler/compat.go

```diff
@@ -74,3 +78,51 @@ func engineQueryFunc(engine *promql.Engine, q storage.Queryable, delay time.Dura
 		return orig(ctx, qs, t.Add(-delay))
 	}
 }
+
+// TenantManagerFunc is a function adapter for the TenantManager interface
+type TenantManagerFunc func(
```

Comment:
Let's just use the function directly, without having an interface for it?
I'd also suggest calling it `RulerManagerFactory` to better describe its intent.

Reply:
Fair point. I'll remove it :)